IFComp/lib/IFComp/Controller/Entry.pm

```diff
@@ -238,7 +240,18 @@ sub _build_withdrawal_form {
 }
 
 sub _process_form {
-    my ( $self, $c ) = @_;
+    my ( $self, $c, $action ) = @_;
+
+    my $entcount = $c->model('IFCompDB::Entry')->search(
+        {   author => $c->user->get_object->id,
+            comp   => $c->stash->{current_comp}->id,
+        }
+    )->count;
+    my $entry = $c->stash->{entry};
+
+    if ( $action == $IS_CREATE && $entcount >= $MAX_ENTRIES ) {
+        return 0;
```

Comment:
Could you please add a user-visible form error in this case? I think that `$self->form->add_form_error( ... )` should do the trick. (See https://metacpan.org/pod/distribution/HTML-FormHandler/lib/HTML/FormHandler/Manual/Errors.pod#Form-methods)